api/loader/fileloader.go

```diff
@@ -13,11 +12,36 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/pkg/errors"
+
 	"sigs.k8s.io/kustomize/api/ifc"
 	"sigs.k8s.io/kustomize/api/internal/git"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
 )
 
+// DemandDirectoryRoot assures that the given path is in fact a directory.
+func DemandDirectoryRoot(
+	fSys filesys.FileSystem, path string) (filesys.ConfirmedDir, error) {
+	if path == "" {
+		return "", filesys.ErrEmptyDir
+	}
+	d, f, err := fSys.CleanedAbs(path)
+	if err != nil {
+		return "", errors.Wrapf(err, filesys.ErrCleanAbs.Error())
+	}
+	if f != "" {
+		return "", filesys.ErrFileNotDir
+	}
+	return d, nil
+}
+
+// HasRemoteFileScheme returns whether path has a url scheme that kustomize allows for
+// remote files. See examples/remoteBuild.md
+func HasRemoteFileScheme(path string) bool {
+	u, err := url.Parse(path)
+	return err == nil && (u.Scheme == "http" || u.Scheme == "https")
```

Comment:
Can we also use this new function here? https://github.com/kubernetes-sigs/kustomize/blob/37a403fc65d417e591ee874d48081cac7986f513/api/loader/fileloader.go#L305

It will help us be consistent across both `kustomize localize` and `kustomize build`

Reply:
> Can we also use this new function here?

[kustomize/api/loader/fileloader.go](https://github.com/kubernetes-sigs/kustomize/blob/37a403fc65d417e591ee874d48081cac7986f513/api/loader/fileloader.go#L305)

Line 305 in [37a403f](https://github.com/kubernetes-sigs/kustomize/commit/37a403fc65d417e591ee874d48081cac7986f513)

 if u, err := url.Parse(path); err == nil && (u.Scheme == "http" || u.Scheme == "https") { 
It will help us be consistent across both kustomize localize and kustomize build

I think I did replace that line with the function [here in the PR](https://github.com/kubernetes-sigs/kustomize/pull/4652/files#diff-f0497a54c02ff8286aad3ea21c102338afadf07571e70fccfcc995b83f72d066R331). But lmk if I'm misunderstanding something.

Reply:
> And it would be a good idea to locally run all the tests you added in #4615 if/when you make this change.
Just to make sure, you're recommending I locally run the tests in #4615 that are currently skipped. Because we know the others pass.

Reply:
> I think I did replace that line with the function here in the PR. But lmk if I'm misunderstanding something.

Ah thanks! Must have missed that

> Just to make sure, you're recommending I locally run the tests in #4615 that are currently skipped. Because we know the others pass.

Yup! Just to verify that they still pass locally.